### PR TITLE
I don't always get new commands, but when I do, it's a MEME command.

### PR DESCRIPTION
**Pull request by PureKrome**
- Added new command: /meme.

I don't always get new commands, but when I do, it's a MEME command.

### CODE REVIEW

File: JabbR/Commands/MemeCommand.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+                throw new HubException(LanguageResources.Meme_DataRequired);
+            }
+
+            if (args.Length != 3)
+            {
+                throw new HubException(LanguageResources.Meme_IncorrectNumberOfArguments);
+            }
+
+            ChatRoom callingRoom = context.Repository.GetRoomByName(callerContext.RoomName);
+            string message = String.Format("http://upboat.me/{0}/{1}/{2}.jpg", args[0], args[1], args[2]);
```
Comment (davidfowl) [inline JabbR/Commands/MemeCommand.cs]: Is there an https version?

Comment (DavidChristiansen) [inline JabbR/Commands/MemeCommand.cs]: Doesn't appear so Dave... :-1:

**Comment by PureKrome** [comment]
I've [created a new issue](https://github.com/worldwidewat/upboat.me/issues/52) asking those people to hopefully buy an SSL cert. they are only around $20 or something...

> Should be space separated ("type top-line bottom-line")

EDIT: (my bad)

@cmaish  I'll double check it and make it that.

**Comment by PureKrome** [comment]
@cmaish well spotted, sir! I've fixed up the typos. The command was already correctly doing what u suggested .. so i just fixed the command args-text.

**Comment by mharen** [comment]
Hi, I'm one of the guys behind upboat.me. SSL support just landed over there. Let me know if there's anything else I might be able to help with.

**Comment by PureKrome** [comment]
a

b

File: JabbR/Commands/MemeCommand.cs (hunk 1 above)
Comment (samandmoore) [inline JabbR/Commands/MemeCommand.cs]: so no @PureKrome needs to update this to use the https url :)

File: JabbR/Hubs/Chat.cs [hunk 2]
```diff
@@ -1056,6 +1056,11 @@ void INotificationService.ChangeWelcome(ChatUser user, ChatRoom room)
             Clients.User(user.Id).welcomeChanged(isWelcomeCleared, parsedWelcome);
         }
 
+        void INotificationService.GenerateMeme(ChatUser user, ChatRoom room, string message)
+        {
+            Send(message, room.Name);
```
Comment (davidfowl) [inline JabbR/Hubs/Chat.cs:1061]: Send?

Comment (PureKrome) [inline JabbR/Hubs/Chat.cs:1061]: Yep - I didn' tknow what was better / appropriate.

Comment (davidfowl) [inline JabbR/Hubs/Chat.cs:1061]: most of the other commands have specific methods

**Comment by mharen** [comment]
@PureKrome, it was mentioned in the chat room the other day that an API for the meme list would be useful if anyone ever wanted to implement tab-completion of meme names.

I [added](https://github.com/worldwidewat/upboat.me/issues/53) that api here:
- http://upboat.me/api/list

Hopefully that's helpful :)

**Pull request merged**